src/Documentation/Advanced-Concepts/StatelessWorker.md

```diff
@@ -0,0 +1,84 @@
+This this the most intuitive expression of the Virtual Actor model with each grain corresponding to an entity with a unique type/identity.
+However, there are also cases when an application needs to perform functional stateless operations that are not tied to a particular entity in the system.
+For example, if client sends requests with encrypted payloads that need to be decrypted before they could be routed to the target grain for processing, such decryption/routing logic is not tied to a specific entity in the application, and can easily scale out.
+
+When the `[StatelessWorker]` attribute is applied to a grain class, it indicates to the Orleans runtime grains of that class should be treated as **Stateless Worker** grains.
+**Stateless Worker** grain have the following properties that make their execution very different from that of normal grain classes.
+
+1. The Orleans runtime can and will create multiple activations of a Stateless Worker grain on different silos of the cluster.
+2. Requests made to Stateless Worker grains are always executed locally, that is on the same silo where the request originated, either made by a grain running on the silo or received by the silo's client gateway.
+Hence, calls to Stateless Worker grains never incur a remote message.
+3. The Orleans Runtime automatically creates additional activations of a Stateless Worker grain if the already existing ones are busy.
+The maximum number of activations of a Stateless Worker grain the runtime creates per silo is limited by default by the number of CPU cores on the machine, unless specified explicitly by the optional `maxLocalWorkers` argument.
+4. Because of 2 and 3, Stateless Worker grain activations are not individually addressable. Two subsequent requests to a Stateless Worker grain may be processed by different activations of it.
+
+Stateless Worker grains provide a straightforward way of creating an auto-managed pool of grain activations that automatically scales up and down based on the actual load.
+The runtime always scans for available Stateless Worker grain activations in the same order.
+Because of that, it always dispatches a requests to the first idle local activation it can find, and only gets to the last one if all previous activations are busy.
+If all activations are busy and the activation limit hasn't been erached, it creates one more activation at the end of the list, and dispatches the request to it.
+That means that when the rate of requests to a Stateless Worker grain increases, and existing activations get all busy, the runtime expands the pool of its activations up to the limit.
+Conversely, when the load drops, and it can be handled by a smaller number of activations of the Stateless Worker grain, the activations at the tail of the list will not be getting requests dispatched to them.
+They will become idle, and eventually deactivated by the standard activation garbage collection process.
```

Comment:
I think we should shy away from the term "garbage collection": it implies that we're throwing the grain away. We know that's not the case, but users typically use the words "activation" & "grain" interchangeably and they probably don't want to lose their state. Maybe "activation collection" is less likely to cause confusion. I'm happy either way.